bionic/code_analyzer.py

```diff
@@ -0,0 +1,134 @@
+"""
+This module implements the logic to analyze your code objects and
+returns the result in the form of a byte array.
+"""
+
+import inspect
+import json
+import warnings
+
+from .utils.misc import oneline
+
+
+DEFAULT_NAME = b"zA)T+ndu/Ygd%JH=eux)[F9:<>)zND]}u4x,c49tWH<yd5+#+4Xg{=J<[S5/'XG+"
+
+
+def analyze_func(func):
+    assert inspect.isroutine(func)
+
+    analyzer = CodeAnalyzer()
+    analyzer.analyze(func)
+    return analyzer.get_analyzed_bytes()
+
+
+class CodeAnalyzer:
+    """
+    This class contains the logic to analyze a code object. For now, it
```

Comment:
"Analyze" doesn't really say anything about what this class does. What is it useful for?

Reply:
I added more information in the module and changed the language. Lmk if you think I should add more information.